dimod/discrete/discrete_quadratic_model.py

```diff
@@ -27,6 +27,9 @@
 from dimod.sampleset import as_samples
 from dimod.serialization.fileview import VariablesSection, _BytesIO
 from dimod.variables import Variables
+from typing import List, Tuple, Union, Generator
+
+Linear = Union[List[Tuple], Generator[Tuple, None, None]]
```

Comment:
One last thing, I think `Linear` is too broad of a name, there are a lot of other things we'd like to call Linear. Maybe `LinearConstraint` or `LinearTriplets` or something